bundles/mapping/mapmodule/domain/AbstractLayer.js

```diff
@@ -140,6 +140,14 @@ Oskari.clazz.define(
         me.tilesToLoad = 0;
         me.errors = 0;
 
+        me._lastFrameLoadTime = 0;
+        me._scheduleNextTimestep = false;
+        me._currentTime = null;
+        me._mapModule = Oskari.getSandbox().findRegisteredModuleInstance('MainMapModule');
```

Comment:
Mapmodule should not be references here. At the very least it should be passed as parameter, but I don't think that's a good idea either. Maybe move the functionality to mapmodule or a plugin instead and keep the logic out of the domain object.